src/tui.rs

```diff
@@ -369,6 +409,92 @@ impl Tui {
 		}
 	}
 
+	/// Map a terminal cell click (col, row) to a displayed page and MuPDF coordinates.
+	/// Returns (page_index, mupdf_x_px, mupdf_y_px) if the click hits a page image.
+	pub fn map_click_to_page(
+		&self,
+		col: u16,
+		row: u16,
+		full_layout: &RenderLayout,
+		font_size: FontSize
+	) -> Option<(usize, f32, f32)> {
+		let img_area = full_layout.page_area;
+
+		// First, compute how many pages will be shown and their total width using the same
+		// greedy logic as `render` so our positioning matches exactly.
+		let mut total_width: u16 = 0;
+		let mut pages_shown: usize = 0;
+
+		for (idx, page) in self.rendered[self.page..].iter().enumerate() {
+			// only consider pages that have an image ready
+			if page.img.is_none() {
+				break;
+			}
+
+			if let Some(max) = self.page_constraints.max_wide {
+				if idx >= max.get() {
+					break;
+				}
+			}
+
+			let width = page.img.as_ref().map(|img| img.w_h().0).unwrap_or(0);
+
+			match total_width.checked_add(width) {
+				Some(new_total) if new_total <= img_area.width => {
+					total_width = new_total;
+					pages_shown += 1;
+				}
+				_ => break,
+			}
+		}
+
+		if pages_shown == 0 {
+			return None;
+		}
```

Comment:
I think we should be able to trust `self.last_render` to still be correct, so we don't need to recompute all of this - we can just pull the `pages_shown` and `total_width` info directly from that field for this.